pyqtorch/circuit.py

```diff
@@ -207,3 +207,25 @@ def dagger(self, values: dict[str, Tensor]) -> Tensor:
 
     def jacobian(self, values: dict[str, Tensor]) -> Tensor:
         return values[self.param_name] * ones_like(self.unitary(values))
+
+
+def hea(n_qubits: int, n_layers: int, param_name: str) -> tuple[ModuleList, ParameterDict]:
+    def _idx() -> Generator[int, Any, NoReturn]:
+        i = 0
+        while True:
+            yield i
+            i += 1
+
+    def idxer() -> Generator[int, Any, None]:
+        yield from _idx()
+
+    idx = idxer()
+    ops = []
+    for _ in range(n_layers):
+        layer = []
+        for i in range(n_qubits):
+            layer += [Merge([fn(i, f"{param_name}_{next(idx)}") for fn in [RX, RY, RX]])]
+        ops += layer
+        ops += [Sequence([CNOT(i % n_qubits, (i + 1) % n_qubits) for i in range(n_qubits)])]
+    params = ParameterDict({f"theta_{n}": rand(1, requires_grad=True) for n in range(next(idx))})
+    return ops, params
```

Comment:
I would add this to a `constructors` file

Reply:
i will add this in the next MR